contracts/lib/OrderCombiner.sol

```diff
@@ -175,6 +180,32 @@ contract OrderCombiner is OrderFulfiller, FulfillmentApplier {
             mstore(orderHashes, 0)
         }
 
+        // Declare an error buffer indicating status of any native offer items.
+        // {00} == 0 => In a match function, no native offer items: allow.
+        // {01} == 1 => In a match function, some native offer items: allow.
+        // {10} == 2 => Not in a match function, no native offer items: allow.
+        // {11} == 3 => Not in a match function, some native offer items: THROW.
+        uint256 invalidNativeOfferItemErrorBuffer;
+
+        // Use assembly to set the value for the second bit of the error buffer.
+        assembly {
+            // Use the second bit of the error buffer to indicate whether the
+            // current function is not matchAdvancedOrders or matchOrders.
+            invalidNativeOfferItemErrorBuffer := shl(
```

Comment:
Likely no need for assembly on this and it clearly shows how risky it is to the reader:
```solidity
unchecked {
  invalidNativeOfferItemErrorBuffer = ((uint32(msg.selector) % NonMatchSelector_MagicModulus) > NonMatchSelector_MagicRemainder) ? 3 : 0;
}
```

Reply:
I do like including this as part of the documentation to demonstrate what’s happening here!